src/commands/mod.rs

```diff
@@ -1009,6 +1032,229 @@ impl DaemonControl {
+    ///
+    /// If the previous transaction includes a change output to one of our own change addresses,
+    /// this same address will be used for change in the RBF transaction, if required. If the previous
+    /// transaction pays to more than one of our change addresses, then the one receiving the highest
+    /// value will be used as a change address and the others will be treated as non-change outputs.
+    ///
+    /// `is_cancel` indicates whether to "cancel" the transaction by removing non-change outputs from
+    /// the replacement or otherwise to keep the same non-change outputs and simply bump the fee.
+    /// If `true`, the only output of the RBF transaction will be change and the inputs will include
+    /// at least one of the inputs from the previous transaction. If `false`, all inputs from the previous
+    /// transaction will be used in the replacement.
+    /// In both cases, the RBF transaction may include additional confirmed coins as inputs if required
+    /// in order to pay the higher fee (this applies also when replacing a self-send).
+    ///
+    /// `feerate_vb` is the target feerate for the RBF transaction (in sat/vb). If `None`, it will be set
+    /// to 1 sat/vb larger than the feerate of the previous transaction, which is the minimum value allowed
+    /// when using RBF.
+    pub fn rbf_psbt(
+        &self,
+        txid: &bitcoin::Txid,
+        is_cancel: bool,
+        feerate_vb: Option<u64>,
+    ) -> Result<CreateSpendResult, CommandError> {
+        // The transaction to be replaced must be one of ours and so we can assume it signals replaceability.
+        // The inputs in the replacement transaction will be either those coins used as inputs in the transaction
+        // to be replaced or "confirmed" coins. Therefore, we can assume the transaction to be replaced is
+        // the only one directly conflicting with the replacement transaction.
+        let mut db_conn = self.db.connection();
+
+        let prev_psbt = db_conn
+            .spend_tx(txid)
+            .ok_or(CommandError::UnknownSpend(*txid))?;
+        let prev_coins = {
+            let prev_outpoints: &Vec<bitcoin::OutPoint> = &prev_psbt
+                .unsigned_tx
+                .input
+                .iter()
+                .map(|txin| txin.previous_output)
+                .collect();
+            let prev_coins = db_conn.coins_by_outpoints(prev_outpoints);
+            assert_eq!(prev_outpoints.len(), prev_coins.len());
+            prev_coins
+        };
+        if prev_coins
+            .iter()
+            .any(|(_, coin)| coin.spend_block.is_some())
+        {
+            return Err(CommandError::RbfError(RbfErrorInfo::TxAlreadyConfirmed(
+                *txid,
+            )));
+        }
+        let mempool_entry = self
+            .bitcoin
+            .mempool_entry(txid)
+            .ok_or(CommandError::RbfError(RbfErrorInfo::TxNotYetBroadcast(
+                *txid,
+            )))?;
```

Comment:
That sounds good. Will change it to handle if tx is not in the mempool.

Reply:
I don't think there is a way to distinguish both.

Good point. I had overlooked that. You can use `gettxspendingprevout` to list the transactions spending a given set of coins. This can be a more generalistic way of getting the target min fee: always use `gettxspendingprevout` and query `getmempoolentry` for each of them. If the list is empty, assume it was purged and use a minfee of 0?

Reply:
Would it be possible to look up the `spend_txid` for each coin in the DB instead of using `gettxspendingprevout`?

Reply:
I've updated it now to get spending transactions from the DB.

Even if it's not in the DB or mempool, I always include the `txid` of the transaction to be replaced in the set of directly conflicting transactions so that the replacement's feerate will be higher. If the transaction to be replaced is not in the mempool, I make the assumption that the PSBT has been finalized so that we can get its fee and vsize, but perhaps I need to check this is the case as it's now possible to call the `rbfpsbt` command on an unsigned PSBT. I've added comments regarding this in the code.